### ISSUE TITLE
KeyCorridor: match MiniGrid's board distribution while keeping static shapes

### ISSUE THREAD

**Issue by xweinp** [epignatelli/navix#223]
_Posted by an AI agent (Claude Opus 5, via Claude Code) on behalf of @xweinp._

## Summary

#162 ported MiniGrid's `RoomGrid.connect_all` into `KeyCorridor._reset` and made two deliberate simplifications, both to keep the reset a fixed-shape `jax.jit` program:

1. **A randomised Kruskal pass instead of MiniGrid's retry loop.** A wall is opened only if it joins two components, because the sample-with-replacement loop has no static iteration bound. The PR describes this as giving the same reachability guarantee.
2. **Every candidate wall stays a `Door` entity**, so the entity count is fixed. The unused ones get `requires=-2`, so they can never be opened. The PR and its review both note that these render as locked doors where MiniGrid draws a wall, and call that an unavoidable cost of static shapes.

Both trade-offs were reasonable, but neither is actually forced by `jax.jit`. This issue proposes matching MiniGrid's board distribution exactly while keeping static shapes. Two further differences (3 and 4 below) date from the original KeyCorridor port and were never discussed.

I compared 3000 resets per registered size from each library: MiniGrid 3.1.0 on seeds 0–2999, and NAVIX 4.0.0 on `jax.random.split(PRNGKey(0), 3000)`.

### 1. Unused candidate walls can render as walls without changing shapes

Currently every S4R3 board shows 10 doors: 1 key-locked, 5 unlocked, and 4 with `requires=-2` that are drawn as locked doors. MiniGrid shows 6–8 doors and plain wall everywhere else.

The fixed entity count can be kept by moving unused candidates to `DISCARD_PILE_COORDS` and leaving their grid cell a wall. That is the repo's existing convention for off-grid entities: picked-up items, `ObstructedMaze`'s boxed keys, `Unlock`. `rgb`, `symbolic` and `categorical` already skip entities at that position.

### 2. The Kruskal pass keeps reachability but not MiniGrid's door count

`connect_all` adds a door on every eligible wall it draws until the rooms connect, including walls between rooms that are already connected. The Kruskal pass always builds a spanning tree:

| unlocked doors | MiniGrid S3R2 | NAVIX S3R2 | MiniGrid S4R3 | NAVIX S4R3 |
|---|---|---|---|---|
| 3 | 0.754 | 1.000 | | |
| 4 | 0.246 | 0 | | |
| 5 | | | 0.409 | 1.000 |
| 6 | | | 0.362 | 0 |
| 7 | | | 0.230 | 0 |

The count depends only on the room graph, so S3R3, S5R3 and S6R3 match S4R3 to within sampling noise.

The retry loop can be ported exactly in the same fixed-length pass. Each wall is equally likely to be drawn, from either side, and walls that already have a door are skipped. So the order in which distinct eligible walls are *first* drawn is a uniform random permutation. The doors `connect_all` adds are exactly the shortest prefix of that permutation that connects the rooms. The existing permutation and union-find already compute this. The only change is the condition: add a door while more than one component is left, rather than only when the wall merges two components.

### 3. The agent starts in a random corridor row

MiniGrid calls `place_agent(1, num_rows // 2)`. The agent always starts in the middle room of the corridor, on any free cell of that room's box, which includes the corridor openings in its top and bottom walls. A pose facing the locked door is redrawn. NAVIX picks a random corridor row, uses interior cells only, and can face the locked door. This dates from the original port.

### 4. `RoomsGrid.position_in_room` never reaches the last interior row or column

```python
local_row = jax.random.randint(k1, (), minval=1, maxval=self.room_size[0])
```

`maxval` is exclusive, so the last interior row and column are never drawn. `position_on_border` right below it uses `room_size + 1`. For room sizes of 2 and up (S4R3 and larger), the key and goal never land on a room's bottom or right interior edge. KeyCorridor is the only caller.

## Reproduction

```python
import jax, navix as nx, numpy as np

env = nx.make("Navix-KeyCorridorS4R3-v0")
state = jax.vmap(env.reset)(jax.random.split(jax.random.PRNGKey(0), 4)).state
print(np.asarray(state.entities["door"].requires))
print(np.asarray(state.entities["player"].position)[:, 0])
```

```
[[-2 -1 -2  1 -1 -1 -1 -2 -1 -2]
 [-2  1 -1 -2 -1 -1 -1 -2 -2 -1]
 [-1 -1 -1  1 -1 -1 -2 -2 -2 -2]
 [-1  1 -1 -2 -2 -1 -2 -2 -1 -1]]
[[7 4]
 [7 4]
 [1 4]
 [4 4]]
```

Every reset has exactly five `-1` doors and four `-2` doors. The agent starts in rows 7, 1 and 4, which are three different corridor rooms.

## Not in scope

These also differ from MiniGrid but concern the task rather than the board:

- MiniGrid's target is a `Ball` picked up with `pickup`; here it is a walkable `Goal`, which #126 chose on purpose.
- MiniGrid does not use up the key when it opens a door.
- The reward shaping and `max_steps` differ.
